miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java

```diff
@@ -605,6 +609,57 @@ public boolean processNotification(Run source, int laneCount, String containerSe
     return isNew;
   }
 
+  private boolean updateMetricsFromNotification(Run source, Run target) {
+    if (source.getMetrics() != null && source.getMetrics().equals(target.getMetrics())) return false;
+    if (source.getMetrics() == null && target.getMetrics() != null) {
+      return false;
```

Comment:
Can probably return false if source metrics are null, regardless of target metrics?